lib/git-fastclone.rb

```diff
@@ -189,18 +189,39 @@ def parse_inputs
       [url, path, options]
     end
 
+    def clear_clone_dest_if_needed(attempt_number, clone_dest)
+      return unless attempt_number.positive?
+
+      clone_dest_contents = Dir["#{clone_dest}/*"]
+      return if clone_dest_contents.empty?
+
+      clear_clone_dest(clone_dest_contents)
+    end
+
+    def clear_clone_dest(clone_dest_contents)
+      puts 'Non-empty clone directory found, clearing its content now.'
+      FileUtils.rm_rf(clone_dest_contents)
+    end
+
     # Checkout to SOURCE_DIR. Update all submodules recursively. Use reference
     # repos everywhere for speed.
     def clone(url, rev, src_dir, config)
+      clone_dest = File.join(abs_clone_path, src_dir).to_s
       initial_time = Time.now
 
-      with_git_mirror(url) do |mirror|
+      if Dir.exist?(clone_dest) && !Dir.empty?(clone_dest)
+        raise "Can't clone into an existing non-empty path: #{clone_dest}"
+      end
+
+      with_git_mirror(url) do |mirror, attempt_number|
+        clear_clone_dest_if_needed(attempt_number, clone_dest)
```

Comment:
IIUC this is just a safety check in case the directory got populated for some reason after the check above?
```
if Dir.exist?(clone_dest) && !Dir.empty?(clone_dest)
```
If it didn't raise above I'd expect this dir to be non-existent or empty unless I'm missing something?

I guess my Q is in which scenarios `clear_clone_dest_if_needed` will run `clear_clone_dest`?

Reply:
Nvm just re-read your PR description. Sorry for missing that!

Reply:
No worries, anything left to clarify?